composer.json

```diff
@@ -86,7 +86,7 @@
         "monolog/monolog": "~1.11",
         "ocramius/proxy-manager": "~0.4|~1.0|~2.0",
         "symfony/phpunit-bridge": "~3.2",
-        "egulias/email-validator": "~1.2,>=1.2.1",
+        "egulias/email-validator": "~1.2,>=1.2.1,<2.0",
```

Comment:
`~1.2` already includes a restriction to <2.0. But as this is the dev requirement, it does not apply for projects using Symfony.
What you need to do is adding a conflict rule (to make it conflict with `>=2.0`), both in symfony/symfony and symfony/validator.